### DISCUSSION TITLE
Need help configuring Hyprland to turn off display instead of hibernating when closing laptop lid

### DISCUSSION THREAD

**Discussion by st-hh** [hyprwm/Hyprland#4842] [Q&A]
Hey everyone,

I'm new to Hyprland and seeking some assistance with configuring it on my laptop. I'm trying to set it up so that when I close the laptop lid, the display turns off instead of the laptop going into sleep/hibernation mode since it's connected to external display.

I've tried followed the instructions on the Hyprland Wiki here: [link](https://wiki.hyprland.org/Configuring/Binds/#switches), specifically:

1. Added the following commands to my Hyprland configuration file:

```plaintext
bindl=,switch:off:Lid Switch,exec,hyprctl keyword monitor "eDP-1, preferred, auto, auto"
bindl=,switch:on:Lid Switch,exec,hyprctl keyword monitor "eDP-1, disable"
```

2. Modified the `/etc/systemd/logind.conf` file by uncommenting the `HandleLidSwitch` line and changing its value to `ignore`:

```plaintext
HandleLidSwitch=ignore
```

3. Restarted the systemd service:

```bash
sudo systemctl restart systemd-logind
```

Despite these changes, the laptop still goes into hibernation mode when I close the lid.

Could anyone please help me figure out what I might be doing wrong or suggest alternative configurations to achieve this?

Any assistance would be greatly appreciated. Thanks in advance!

**Comment by Swarnim114** [comment]
it wasnt working on my system too 
i just binded a new bind to lock the screen then making it sleep 
bind = $mainMod+Shift,Q , exec, bash -c "swaylock && systemctl suspend" # lock screen and then suspend
maybe this is what you need

**Comment by kayaefe** [comment]
Hey, were you able to come up with a solution to this problem?
I have taken the same steps with no luck.

**Comment by theblack-don** [comment]
@st-hh Did you ever get this one figured out? I am also having trouble accomplishing this

**Comment by reddcarp** [comment]
I used the following line in my hyprland config:

```
bindl = , switch:on:Lid Switch, exec, hyprctl dispatch dpms off
bindl = , switch:off:Lid Switch, exec, hyprctl dispatch dpms on
```

> Note: This affects all displays connected

I would suggest looking at the comment [here](https://wiki.hyprland.org/Configuring/Monitors/#disabling-a-monitor) which explains what the `monitor disable`  instruction actually do (as I believe it isn't what is desired in the use case described above).

Look here for an explanation of what DPMS is: https://tldp.org/HOWTO/Battery-Powered/displaytypes.html

**Comment by harshv5094** [comment]
In my case, I'm using it with `hyprlock`.
```conf
bindl=,switch:on:Lid Switch, exec, hyprctl dispatch dpms off && hyprctl dispatch exec hyprlock
bindl=,switch:off:Lid Switch, exec, hyprctl dispatch dpms on
```

**Comment by msmafra** [comment]
Probably already solved

In my case, on Fedora, there is no logind.conf file in any of the locations, nor I a template to uncomment lines, so I created one the folder _/etc/systemd/logind.conf.d_  and put a file named _60-logind-lid-switch.conf_ inside with:
```
[Login]
HandleLidSwitch=suspend
HandleLidSwitchExternalPower=ignore
HandleLidSwitchDocked=ignore
```
With those lines my laptop only disables the monitor instead of suspending it when plugged in or docked (I don't have a away to test this one), but will suspend if on battery. If you want to for all the sittuations, change HandleLidSwitch to "ignore" too.
On Fedora there is another step that make it work, needed because of SELinux, before reloading or restarting the service. 

```
doas restorecon -F -R /etc/systemd
doas systemctl reload systemd-logind.service
```

After testing with **restart** and **reload**, **reload** worked well with no session problems since **restart** would cause problems on my system, making me have to reboot it to make the new configuration work.

I took the info from here
https://discussion.fedoraproject.org/t/fedora-server-40-issue-preventing-suspend-on-lid-close/119185/2
